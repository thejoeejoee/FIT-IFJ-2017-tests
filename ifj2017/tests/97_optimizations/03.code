' simple constant propagation

scope
	dim const as integer = 25

	print const; !"\n";
	print const * 2; !"\n";
	print const * 2 + const; !"\n";
	print 2 / const; !"\n";
	print const * (const * const + 3); !"\n";
	print (const - 98) / (const + 6) * (const * const + 3); !"\n";

end scope