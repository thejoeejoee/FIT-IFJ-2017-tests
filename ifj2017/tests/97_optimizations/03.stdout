 25
 50
 75
 0.08
 15700
-1478.84
